crates/accelerate/src/split_2q_unitaries.rs

```diff
@@ -19,20 +21,77 @@ use qiskit_circuit::circuit_instruction::OperationFromPython;
 use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType, Wire};
 use qiskit_circuit::imports::UNITARY_GATE;
 use qiskit_circuit::operations::{Operation, Param};
+use qiskit_circuit::packed_instruction::PackedInstruction;
+use qiskit_circuit::Qubit;
 
 use crate::two_qubit_decompose::{Specialization, TwoQubitWeylDecomposition};
 
+/// Extracts the K1r and K1l gates from the decomposition
+/// and creates them as new 1-qubit unitary gates.
+fn create_k1_gates<'a>(
+    decomp: &'a TwoQubitWeylDecomposition,
+    py: Python<'a>,
+) -> PyResult<(Bound<'a, PyAny>, Bound<'a, PyAny>)> {
+    let k1r_arr = decomp.K1r(py);
+    let k1l_arr = decomp.K1l(py);
+    let kwargs = PyDict::new(py);
+    kwargs.set_item(intern!(py, "num_qubits"), 1)?;
+    let k1r_gate = UNITARY_GATE
+        .get_bound(py)
+        .call((k1r_arr, py.None(), false), Some(&kwargs))?;
+    let k1l_gate = UNITARY_GATE
+        .get_bound(py)
+        .call((k1l_arr, py.None(), false), Some(&kwargs))?;
+    Ok((k1r_gate, k1l_gate))
+}
+
+/// Creates a new instruction and adds it to the DAG.
+fn add_new_op(
+    new_dag: &mut DAGCircuit,
+    new_op: OperationFromPython,
+    qargs: Vec<Qubit>,
+    mapping: &[usize],
+    py: Python,
+) -> PyResult<()> {
+    let inst = PackedInstruction {
+        op: new_op.operation,
+        qubits: new_dag.qargs_interner.insert_owned(qargs),
+        clbits: new_dag.cargs_interner.get_default(),
+        params: (!new_op.params.is_empty()).then(|| Box::new(new_op.params)),
+        extra_attrs: new_op.extra_attrs,
+        #[cfg(feature = "cache_pygates")]
+        py_op: std::sync::OnceLock::new(),
```

Comment:
You do have the py op available for caching here since you are generating this via python. But in this case it doesn't actually matter too much because this is always going to be a `PyGate` and if the cache is `None` the python object reference will just be taken from that.

Reply:
So, do you think something needs to change here?

Reply:
I think this is fine as is. This was just more my idle musings.